src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -396,6 +399,120 @@ private void VerifyIsAlive()
                 throw new InvalidOperationException(SR.AssemblyLoadContext_Verify_NotUnloading);
             }
         }
+
+        internal static class StaticAsyncLocalCurrentContextualReflectionContext
```

Comment:
Now that it is not initialized via static constructor, it does not really need to be enclosed in a separate type.